0.9.0

ChamberQin opened the pull request.

gwenshap committed:
Changing version to 0.9.0.0

hachikuji committed:
KAFKA-2723: new consumer exception cleanup (0.9.0)

Author: Jason Gustafson <jason@confluent.io>

Reviewers: Guozhang Wang

Closes #452 from hachikuji/KAFKA-2723

hachikuji committed:
HOTFIX: fix group coordinator edge cases around metadata storage callback (0.9.0)

Author: Jason Gustafson <jason@confluent.io>

Reviewers: Guozhang Wang

Closes #453 from hachikuji/hotfix-group-coordinator-0.9

ewencp committed:
KAFKA-2480: Handle retriable and non-retriable exceptions thrown by sink tasks.

Author: Ewen Cheslack-Postava <me@ewencp.org>

Reviewers: Gwen Shapira

Closes #450 from ewencp/kafka-2480-unrecoverable-task-errors

(cherry picked from commit f4b87deefecf4902992a84d4a3fe3b99a94ff72b)
Signed-off-by: Gwen Shapira <cshapi@gmail.com>

Commit:
KAFKA-2773; 0.9.0 branch)Fixed broken vagrant provision scripts for static zk/broker cluster

Author: Geoff Anderson <geoff@confluent.io>

Reviewers: Gwen Shapira

Closes #455 from granders/KAFKA-2773-0.9.0-vagrant-fix

ewencp committed:
KAFKA-2774: Rename Copycat to Kafka Connect

Author: Ewen Cheslack-Postava <me@ewencp.org>

Reviewers: Gwen Shapira

Closes #456 from ewencp/kafka-2774-rename-copycat

(cherry picked from commit f2031d40639ef34c1591c22971394ef41c87652c)
Signed-off-by: Gwen Shapira <cshapi@gmail.com>

rajinisivaram committed:
KAFKA-2779; Close SSL socket channel on remote connection close

Close socket channel in finally block to avoid file descriptor leak when remote end closes the connection

Author: Rajini Sivaram <rajinisivaram@googlemail.com>

Reviewers: Ismael Juma <ismael@juma.me.uk>, Jun Rao <junrao@gmail.com>

Closes #460 from rajinisivaram/KAFKA-2779

(cherry picked from commit efbebc6e843850b7ed9a1d015413c99f114a7d92)
Signed-off-by: Jun Rao <junrao@gmail.com>

ewencp committed:
KAFKA-2781; Only require signing artifacts when uploading archives.

Author: Ewen Cheslack-Postava <me@ewencp.org>

Reviewers: Jun Rao <junrao@gmail.com>

Closes #461 from ewencp/kafka-2781-no-signing-for-install

(cherry picked from commit a24f9a23a6d8759538e91072e8d96d158d03bb63)
Signed-off-by: Jun Rao <junrao@gmail.com>

ewencp committed:
KAFKA-2776: Fix lookup of schema conversion cache size in JsonConverter.

Author: Ewen Cheslack-Postava <me@ewencp.org>

Reviewers: Gwen Shapira

Closes #458 from ewencp/kafka-2776-json-converter-cache-config-fix

(cherry picked from commit e9fc7b8c84908ae642339a2522a79f8bb5155728)
Signed-off-by: Gwen Shapira <cshapi@gmail.com>

hachikuji committed:
HOTFIX: bug updating cache when loading group metadata

The bug causes only the first instance of group metadata in the topic to be written to the cache (because of the putIfNotExists in addGroup). Coordinator fail-over won't work properly unless the cache is loaded with the right metadata.

Author: Jason Gustafson <jason@confluent.io>

Reviewers: Guozhang Wang

Closes #462 from hachikuji/hotfix-group-loading

(cherry picked from commit 2b04004de878823fe631af1f3f85108c0b38caec)
Signed-off-by: Guozhang Wang <wangguoz@gmail.com>

ewencp committed:
KAFKA-2775: Move exceptions into API package for Kafka Connect.

Author: Ewen Cheslack-Postava <me@ewencp.org>

Reviewers: Gwen Shapira

Closes #457 from ewencp/kafka-2775-exceptions-in-api-package

(cherry picked from commit bc76e6704e8f14d59bb5d4fcf9bdf544c9e463bf)
Signed-off-by: Gwen Shapira <cshapi@gmail.com>

ewencp committed:
KAFKA-2778: Use zero loss settings by default for Connect source producers.

Author: Ewen Cheslack-Postava <me@ewencp.org>

Reviewers: Gwen Shapira

Closes #459 from ewencp/kafka-2778-connect-source-zero-loss-settings

(cherry picked from commit 13ba57dcfc45e60de281cc55125e7446322308ba)
Signed-off-by: Gwen Shapira <cshapi@gmail.com>

ewencp committed:
KAFKA-2782: Fix KafkaBasedLogTest assertion and move it to the main test thread.

Author: Ewen Cheslack-Postava <me@ewencp.org>

Reviewers: Gwen Shapira

Closes #463 from ewencp/kafka-2782-fix-kafka-based-log-test-assertion

(cherry picked from commit 75f2b8c8b42459a3ab2962f75f0e05fbbb9e8333)
Signed-off-by: Gwen Shapira <cshapi@gmail.com>

Commit:
KAFKA-2258: add failover to mirrormaker test

This PR adds failover to simple end to end mirror maker test

Marked as WIP for 2 reasons:
- We may want to add a couple more test cases where kafka is being used to store offsets
- There appears to be a test failure in the hard failover case

Author: Geoff Anderson <geoff@confluent.io>

Reviewers: Ewen Cheslack-Postava

Closes #427 from granders/KAFKA-2258-mirrormaker-test

(cherry picked from commit 7073fa7efea4acd6615f9dad4f649f6094be6332)
Signed-off-by: Gwen Shapira <cshapi@gmail.com>

granthenke committed:
KAFKA-2783; Drop outdated hadoop contrib modules

Author: Grant Henke <granthenke@gmail.com>

Reviewers: Gwen Shapira

Closes #466 from granthenke/drop-contrib

(cherry picked from commit 69af573b35f04657e31f60e636aba19ffa0b2c84)
Signed-off-by: Gwen Shapira <cshapi@gmail.com>

hachikuji committed:
KAFKA-2674: clarify onPartitionsRevoked behavior

Author: Jason Gustafson <jason@confluent.io>

Reviewers: Guozhang Wang

Closes #467 from hachikuji/KAFKA-2674

(cherry picked from commit 359be3a682951fd469d690df8d9e7a5a89a9d03b)
Signed-off-by: Guozhang Wang <wangguoz@gmail.com>

guozhangwang committed:
MINOR: remove old producer in config sections to align with APIs

Author: Guozhang Wang <wangguoz@gmail.com>

Reviewers: Gwen Shapira

Closes #468 from guozhangwang/WikiUpdate

(cherry picked from commit a87b978223a827c34db1341b2da3e17e571c4464)
Signed-off-by: Gwen Shapira <cshapi@gmail.com>

Commit:
MINOR: Improve exception message that gets thrown for non-existent group

Author: Ashish Singh <asingh@cloudera.com>

Reviewers: Guozhang Wang

Closes #471 from SinghAsDev/ExceptionMessage

(cherry picked from commit 3ebb4ee1b0803b755eaefc3286fe107868b7b085)
Signed-off-by: Guozhang Wang <wangguoz@gmail.com>

ewencp committed:
KAFKA-2785; Include Kafka Connect jars in releaseTarGz.

Author: Ewen Cheslack-Postava <me@ewencp.org>

Reviewers: Jun Rao <junrao@gmail.com>

Closes #473 from ewencp/kafka-2785-include-copycat-jars-in-release

(cherry picked from commit 79bdc17ab4b0157ceb5257e6c4ff646a3fa55f6d)
Signed-off-by: Jun Rao <junrao@gmail.com>

ewencp committed:
KAFKA-2379: Add basic documentation for Kafka Connect.

Author: Ewen Cheslack-Postava <me@ewencp.org>

Reviewers: Gwen Shapira

Closes #475 from ewencp/kafka-2379-connect-docs

(cherry picked from commit 83eaf3284fe8535bd06743a367a469d3d01946b1)
Signed-off-by: Gwen Shapira <cshapi@gmail.com>

guozhangwang committed:
MINOR: remove Kafka Streams in 0.9.0

Author: Guozhang Wang <wangguoz@gmail.com>

Reviewers: Jun Rao

Closes #474 from guozhangwang/removeKStream

apovzner committed:
KAFKA-2769; Multi-consumer integration tests for consumer assignment incl. session timeouts and corresponding fixes

-- Refactored multi-consumer integration group assignment validation tests for round-robin assignment
-- Added multi-consumer integration tests for session timeout expiration:
   1. When a consumer stops polling
   2. When a consumer calls close()

-- Fixes to issues found with session timeout expiration tests woth help from Jason Gustafson: Try to avoid  SendFailedException exception by cancelling the scheduled tasks and ensuring metadata update before sending group leave requests + send leave group request with retries.

Author: Anna Povzner <anna@confluent.io>

Reviewers: Jason Gustafson, Guozhang Wang

Closes #472 from apovzner/cpkafka-81

(cherry picked from commit c9264b4c8904bf330b70cd84b433a7a141ec9d0e)
Signed-off-by: Guozhang Wang <wangguoz@gmail.com>

hachikuji committed:
KAFKA-2274: verifiable consumer and integration testing

Author: Jason Gustafson <jason@confluent.io>

Reviewers: Guozhang Wang, Geoff Anderson

Closes #465 from hachikuji/KAFKA-2274

ewencp committed:
KAFKA-2786: Only respond to SinkTask onPartitionsRevoked after the WorkerSinkTask has finished starting up.

Author: Ewen Cheslack-Postava <me@ewencp.org>

Reviewers: Gwen Shapira

Closes #476 from ewencp/kafka-2786-on-partitions-assigned-only-after-start

(cherry picked from commit 590a4616a1030b3175ba5a548b5f1e1b49c323c2)
Signed-off-by: Gwen Shapira <cshapi@gmail.com>

ewencp committed:
KAFKA-2797: Only run rat when in the .git repository since it require s the .gitignore to generate the list of files to ignore

Author: Ewen Cheslack-Postava <me@ewencp.org>

Reviewers: Gwen Shapira

Closes #485 from ewencp/kafka-2797-disable-rat-when-git-missing

(cherry picked from commit fe11488a7787732b6cc2ba306f745ea1f318208c)
Signed-off-by: Gwen Shapira <cshapi@gmail.com>

ewencp committed:
KAFKA-2792: Don't wait for a response to the leave group message when closing the new consumer.

Author: Ewen Cheslack-Postava <me@ewencp.org>

Reviewers: Onur Karaman, Gwen Shapira

Closes #480 from ewencp/kafka-2792-fix-blocking-consumer-close

(cherry picked from commit ae5a5d7c08bb634576a414f6f2864c5b8a7e58a3)
Signed-off-by: Gwen Shapira <cshapi@gmail.com>

ewencp committed:
KAFKA-2798: Use prefixedd configurations for Kafka Connect producer and consumer settings so they do not conflict with the distributed herder's settings.

Author: Ewen Cheslack-Postava <me@ewencp.org>

Reviewers: Gwen Shapira

Closes #486 from ewencp/kafka-2798-conflicting-herder-producer-consumer-configs

(cherry picked from commit 403d89edeaa7808f71c0e7318411c925895210f2)
Signed-off-by: Gwen Shapira <cshapi@gmail.com>

Commit:
MINOR: update system test readme

Author: Geoff Anderson <geoff@confluent.io>

Reviewers: Guozhang Wang

Closes #487 from granders/minor-update-test-readme

(cherry picked from commit 1d884d1f60aec9ec7ea334761bead4c60b13c7a9)
Signed-off-by: Guozhang Wang <wangguoz@gmail.com>

hachikuji committed:
KAFKA-2795: fix potential NPE in GroupMetadataManager.addGroup

Author: Jason Gustafson <jason@confluent.io>

Reviewers: Onur Karaman, Guozhang Wang

Closes #488 from hachikuji/KAFKA-2795

(cherry picked from commit c455e608c1f2c7be6ff0a721f49c1fe3ede0165f)
Signed-off-by: Guozhang Wang <wangguoz@gmail.com>

Parth-Brahmbhatt committed:
KAFKA-2788; Allow specifying principals with comman in ACL CLI.

Author: Parth Brahmbhatt <brahmbhatt.parth@gmail.com>

Reviewers: Jun Rao <junrao@gmail.com>

Closes #489 from Parth-Brahmbhatt/KAFKA-2788

(cherry picked from commit 60c06734b345bbf292773e5b9206282ff3995968)
Signed-off-by: Jun Rao <junrao@gmail.com>

ewencp committed:
KAFKA-2793: Use ByteArrayDeserializer instead of StringDeserializer for keys in ConsoleConsumer with new consumer.

Author: Ewen Cheslack-Postava <me@ewencp.org>

Reviewers: Gwen Shapira

Closes #482 from ewencp/kafka-2793-console-consumer-new-consumer-deserializer

(cherry picked from commit 64a1bfeb9b1b90c94cc62ee2587f9745c850ada3)
Signed-off-by: Gwen Shapira <cshapi@gmail.com>

guozhangwang committed:
KAFKA-2799: skip wakeup in the follow-up poll() call.

Cherry-picked from trunk.

Author: Guozhang Wang <wangguoz@gmail.com>

Reviewers: Jason Gustafson

Closes #490 from guozhangwang/K2799

rajinisivaram committed:
KAFKA-2801; Process any remaining data in SSL network read buffer after handshake

Process any remaining data in the network read buffer in `SslTransportLayer` when `read()` is invoked. On handshake completion, there could be application data ready to be processed that was read into `netReadBuffer` during handshake processing. `read()` is already invoked from `Selector` after handshake completion, but data already read into the `netReadBuffer` was not being processed. This PR adds a check for remaining data and continues with processing data if data is available.

Author: Rajini Sivaram <rajinisivaram@googlemail.com>

Reviewers: Jun Rao <junrao@gmail.com>

Closes #493 from rajinisivaram/KAFKA-2801

(cherry picked from commit ea702c384fcfb5a46c5d1354498e418c5801ed30)
Signed-off-by: Jun Rao <junrao@gmail.com>

guozhangwang committed:
KAFKA-2770: Catch and ignore WakeupException for commit upon closing

Author: Guozhang Wang <wangguoz@gmail.com>

Reviewers: Gwen Shapira, Geoff Anderson, Jason Gustafson

Closes #470 from guozhangwang/K2770

(cherry picked from commit e0098b4567541d1c0cefc6f57ce38f67a9133b5e)
Signed-off-by: Guozhang Wang <wangguoz@gmail.com>

gwenshap committed:
KAFKA-2790: doc improvements

Author: Gwen Shapira <cshapi@gmail.com>

Reviewers: Jun Rao, Guozhang Wang

Closes #491 from gwenshap/KAFKA-2790

(cherry picked from commit a8ccdc6154a1e10982cb80df82e8661903eb9ae5)
Signed-off-by: Gwen Shapira <cshapi@gmail.com>

Commit:
KAFKA2805; RecordAccumulator request timeout not enforced when all brokers are gone

Removed the check for expiring only those batches whose metadata is unavailable. Now the batches will be expired irrespective of whether the leader is available or not, as soon as it reaches the requestimeout threshold.

Author: Mayuresh Gharat <mgharat@mgharat-ld1.linkedin.biz>

Reviewers: Jun Rao <junrao@gmail.com>

Closes #503 from MayureshGharat/kafka-2805

(cherry picked from commit 1cd22ed33f1e1f63d8cb63f68309e5d8f43be1e1)
Signed-off-by: Jun Rao <junrao@gmail.com>

Commit:
MINOR: Do not collect zk persistent data by default

In system tests zookeeper service, it is overkill and space-intensive to collect zookeeper data logs by default. This minor patch turns off default collection.

Author: Geoff Anderson <geoff@confluent.io>

Reviewers: Jun Rao <junrao@gmail.com>

Closes #504 from granders/minor-zk-change-log-collect

(cherry picked from commit df88d3be75396b48a762149af2f4bbcd60fe69b9)
Signed-off-by: Jun Rao <junrao@gmail.com>

junrao committed:
KAFKA-2813; selector doesn't close socket connection on non-IOExceptions

Patched Selector.poll() to close the connection on any exception.

Author: Jun Rao <junrao@gmail.com>

Reviewers: Guozhang Wang <wangguoz@gmail.com>, Gwen Shapira <cshapi@gmail.com>

Closes #501 from junrao/KAFKA-2813

(cherry picked from commit 3fd168d9522d1ad25f5582fbe838cea15bdb525f)
Signed-off-by: Jun Rao <junrao@gmail.com>

ijuma committed:
MINOR: Remove `InvalidReceiveException` catch in `SocketServer`

`Selector.poll` no longer throws it.

Author: Ismael Juma <ismael@juma.me.uk>

Reviewers: Jun Rao <junrao@gmail.com>

Closes #506 from ijuma/poll-no-longer-throws-invalid-receive-exception

(cherry picked from commit b5401f291a16977992c50f0a66d9f02aea07bbe7)
Signed-off-by: Jun Rao <junrao@gmail.com>

ijuma committed:
MINOR: Fix logging message in `NetworkClient.poll` not to mention `producer`

Author: Ismael Juma <ismael@juma.me.uk>

Reviewers: Jun Rao <junrao@gmail.com>

Closes #507 from ijuma/fix-error-message-in-network-client-poll

(cherry picked from commit d60e2ea581289a02efcef3b692bbb043fbdf3199)
Signed-off-by: Jun Rao <junrao@gmail.com>

ijuma committed:
MINOR: Use Kafka artifact compiled with Scala 2.11 in quickstart.html

Author: Ismael Juma <ismael@juma.me.uk>

Reviewers: Jun Rao <junrao@gmail.com>

Closes #505 from ijuma/scala-2.11-in-quickstart

(cherry picked from commit 2cf6218849a3634badaf70809923a31934182794)
Signed-off-by: Jun Rao <junrao@gmail.com>

hachikuji committed:
TRIVIAL: provide clearer error in describe group when group is inactive

Author: Jason Gustafson <jason@confluent.io>

Reviewers: Guozhang Wang

Closes #502 from hachikuji/trivial-consumer-groups-fix

(cherry picked from commit d9a455243e2350a18b5c734f0a6edaeb74ac6698)
Signed-off-by: Guozhang Wang <wangguoz@gmail.com>

Commit:
KAFKA-2690: Hide passwords while logging the config.

Added PASSWORD_STRING in ConfigDef that returns "[hidden]" when method toString is invoked.

Author: Jakub Nowak <jakub.nowak94@interia.pl>

Reviewers: Ismael Juma, Gwen Shapira, Jun Rao

Closes #371 from Mszak/ssl-password-protection

(cherry picked from commit ab5ac264a71d7f895b21b4acfd93d9581dabd7c1)
Signed-off-by: Gwen Shapira <cshapi@gmail.com>

ewencp committed:
MINOR: Increase timeouts for Kafka Connect system test service to make tests more reliable on wimpy nodes.

Author: Ewen Cheslack-Postava <me@ewencp.org>

Reviewers: Gwen Shapira

Closes #514 from ewencp/connect-systest-generous-timeouts

Commit:
KAFKA-2822: DescribeConsumerGroup now returns empty list for non-existent group.

…tent group, it used to throw IllegalArgumentException

Author: Ashish Singh <asingh@cloudera.com>

Reviewers: Jason Gustafson, Guozhang Wang

Closes #515 from SinghAsDev/KAFKA-2822

(cherry picked from commit 28d71ce5e06a17248f944147e850f3e5e029bdee)
Signed-off-by: Guozhang Wang <wangguoz@gmail.com>

ewencp committed:
KAFKA-2814: Make Kafka Connect system test REST requests use hostname that is compatible with running under AWS.

Author: Ewen Cheslack-Postava <me@ewencp.org>

Reviewers: Gwen Shapira

Closes #517 from ewencp/kafka-2814-connect-rest-on-aws

(cherry picked from commit 2991c9865d87a715717dbc1bae431a4d39631d9e)
Signed-off-by: Gwen Shapira <cshapi@gmail.com>

Commit:
MINOR: Tuned timeout parameter to reduce chance of transient failure

Increased timeout in downstream consumer doing validation step. This addresses a transient failure case in mirror maker tests with mirror maker failover.

Author: Geoff Anderson <geoff@confluent.io>

Reviewers: Gwen Shapira

Closes #521 from granders/minor-mm-transient-failure

(cherry picked from commit 2802bd081c9cfd91b8d7aa104c145ef0514ada49)
Signed-off-by: Gwen Shapira <cshapi@gmail.com>

guozhangwang committed:
KAFKA-2819; catch NoSuchElementException in ConsoleConsumer

Author: Guozhang Wang <wangguoz@gmail.com>

Reviewers: Jun Rao <junrao@gmail.com>

Closes #516 from guozhangwang/K2819

ijuma committed:
KAFKA-2817; Check if socketChannel is connected in `SslTransportLayer.close`

This avoids spurious log warning messages. Also tweak log message
if wrapResult.getStatus != CLOSED.

Author: Ismael Juma <ismael@juma.me.uk>

Reviewers: Rajini Sivaram <rajinisivaram@googlemail.com>, Jun Rao <junrao@gmail.com>

Closes #511 from ijuma/kafka-2817-unconnected-ssl-transport-layer-close

(cherry picked from commit 528c78fd30b77e8c60e8b3645e7d44404fd4e013)
Signed-off-by: Jun Rao <junrao@gmail.com>

Commit:
Minor: Missing License

Author: Sriharsha Chintalapani <harsha@hortonworks.com>

Reviewers: Gwen Shapira

Closes #524 from harshach/missing-license

(cherry picked from commit 4511aeebc3e37a3d9e2ebeabbd6bbfd16d681719)
Signed-off-by: Gwen Shapira <cshapi@gmail.com>

junrao committed:
trivial doc changes

benstopford committed:
KAFKA-2791; removed deprecated properties

Removed support for BLOCK_ON_BUFFER_FULL_CONFIG (block.on.buffer.full)
Removed support for METADATA_FETCH_TIMEOUT_CONFIG
Removed support for TIMEOUT_CONFIG (aka timeout.ms)

Added support for MAX_BLOCK_MS_CONFIG
Added support for REQUEST_TIMEOUT_MS_CONFIG

Author: Ben Stopford <benstopford@gmail.com>

Reviewers: Jun Rao <junrao@gmail.com>

Closes #525 from benstopford/KAFKA-2791

(cherry picked from commit 4efe4ac6d785c708ec623a6dca24ae34d4b29097)
Signed-off-by: Jun Rao <junrao@gmail.com>

hachikuji committed:
KAFKA-2821; fix deadlock in group metadata write callback

Author: Jason Gustafson <jason@confluent.io>

Reviewers: Ismael Juma <ismael@juma.me.uk>, Jun Rao <junrao@gmail.com>

Closes #519 from hachikuji/KAFKA-2821

(cherry picked from commit 002ec9c796acf8570754522a4eb1b0c3ee9aab13)
Signed-off-by: Jun Rao <junrao@gmail.com>

Commit:
KAFKA-2768: AdminClient ignore member list for non-stable groups

…stabilizing

Author: Ashish Singh <asingh@cloudera.com>

Reviewers: Ismael Juma, Jason Gustafson, Guozhang Wang

Closes #447 from SinghAsDev/KAFKA-2768

(cherry picked from commit a26dbcdf3ac57515cbc9a1c39d8c27e90f74188a)
Signed-off-by: Guozhang Wang <wangguoz@gmail.com>

guozhangwang committed:
KAFKA-2833; print only group offset / metadata according to the formatter

Author: Guozhang Wang <wangguoz@gmail.com>

Reviewers: Jun Rao <junrao@gmail.com>

Closes #527 from guozhangwang/K2833

(cherry picked from commit 43ef0150bdc4ca349af61bf4c75dfb8fd8d14691)
Signed-off-by: Jun Rao <junrao@gmail.com>

junrao committed:
additional trivial 0.9.0 doc changes

Parth-Brahmbhatt committed:
KAFKA-2838; Allow comma in super users, allow comma in CLI authz prop…

…erties.

Author: Parth Brahmbhatt <brahmbhatt.parth@gmail.com>

Reviewers: Sriharsha Chintalapani <harsha@hortonworks.com>, Jun Rao <junrao@gmail.com>

Closes #529 from Parth-Brahmbhatt/KAFKA-2838

(cherry picked from commit 8e6bae21c64699245cf24dbaac3942f32a2a241b)
Signed-off-by: Jun Rao <junrao@gmail.com>

junrao committed:
trivial 0.9.0 doc changes to AclCommand

junrao committed:
trivial 0.9.0 doc changes to fix links

Commit:
KAFKA-2624; Change log message position

Log warning message before truncating log in order to
display right offset value for the truncated log.

Author: Francois Visconte <f.visconte@criteo.com>

Reviewers: Dong Lin <lindong28@gmail.com>, Jun Rao <junrao@gmail.com>

Closes #287 from dopuskh3/KAFKA-2624

(cherry picked from commit 356544caba6448c6ba3bcdb38bea787e1fbc277b)
Signed-off-by: Jun Rao <junrao@gmail.com>

ijuma committed:
KAFKA-2831; Do not use ZKUtils in `ConsumerGroupCommand` if `new-consumer` is used

Author: Ismael Juma <ismael@juma.me.uk>

Reviewers: Jason Gustafson <jason@confluent.io>, Ashish Singh <asingh@cloudera.com>, Jun Rao <junrao@gmail.com>

Closes #528 from ijuma/kafka-2831-consumer-group-command-zookeeper-new-consumer

(cherry picked from commit 5fc4546de7f238a8ee9c6f0b4fe276f0da47707c)
Signed-off-by: Jun Rao <junrao@gmail.com>

granthenke committed:
KAFKA-2809; Improve documentation linking

Often it is useful to link to a specific header within the documentation. Especially when referencing docs in the mailing lists.

This adds anchors and links for all headers in the docs.

Author: Grant Henke <granthenke@gmail.com>

Reviewers: Jun Rao <junrao@gmail.com>

Closes #498 from granthenke/doc-links

(cherry picked from commit 6cbd97597ccf456a4f01f19553da5a03e12c9366)
Signed-off-by: Jun Rao <junrao@gmail.com>

lindong28 committed:
KAFKA-2721; Avoid handling duplicate LeaderAndISR requests

Author: Dong Lin <lindong28@gmail.com>

Reviewers: Jun Rao <junrao@gmail.com?

Closes #436 from lindong28/KAFKA-2721

(cherry picked from commit 6df9e7ff2c6cfb3c7ca16f94928d0e86f3d087e2)
Signed-off-by: Jun Rao <junrao@gmail.com>

junrao committed:
trivial doc change for building customized user name

ewencp committed:
KAFKA-2848; Use client SSL/SASL config utilities in Kafka Connect to avoid duplication of configs.

Author: Ewen Cheslack-Postava <me@ewencp.org>

Reviewers: Jun Rao <junrao@gmail.com>

Closes #539 from ewencp/kafka-2848-reuse-ssl-sasl-client-configs

(cherry picked from commit f1169f1da8728db842aca23dcb6fde740a400699)
Signed-off-by: Jun Rao <junrao@gmail.com>

ijuma committed:
KAFKA-2847; Remove principal builder class from client configs

Also mark `PrincipalBuilder` as `Unstable` and  tweak docs.

Author: Ismael Juma <ismael@juma.me.uk>

Reviewers: Jun Rao <junrao@gmail.com>

Closes #542 from ijuma/kafka-2847-remove-principal-builder-class-from-client-configs

(cherry picked from commit 52d5e88393630ce6f817bd003c7c787e36e31277)
Signed-off-by: Jun Rao <junrao@gmail.com>

Commit:
KAFKA-2746; Add support for using ConsumerGroupCommand on secure install

Author: Ashish Singh <asingh@cloudera.com>

Reviewers: Ismael Juma <ismael@juma.me.uk>, Jason Gustafson <jason@confluent.io>, Jun Rao <junrao@gmail.com>

Closes #534 from SinghAsDev/KAFKA-2746

(cherry picked from commit ffc0965d38c364272078d771fc5ed5f8784e4012)
Signed-off-by: Jun Rao <junrao@gmail.com>

Parth-Brahmbhatt committed:
KAFKA-2852; Updating the Authorizer CLI to use a consistent way to specify a list of values for a config options.

…ecify a list of values for a config options.

Author: Parth Brahmbhatt <brahmbhatt.parth@gmail.com>

Reviewers: Jun Rao <junrao@gmail.com>

Closes #545 from Parth-Brahmbhatt/KAFKA-2852

(cherry picked from commit dacd21ec4e9028593faedeb7d2a4484394367612)
Signed-off-by: Jun Rao <junrao@gmail.com>

guozhangwang committed:
MINOR: fix shutdownHook in ConsoleConsumer

Author: Confluent <confluent@Confluents-MacBook-Pro.local>

Reviewers: Jun Rao <junrao@gmail.com>

Closes #548 from guozhangwang/HFConsoleConsumer

(cherry picked from commit 36b1c1dae292b9a43f56c385de13b89dfd03cad8)
Signed-off-by: Jun Rao <junrao@gmail.com>

ijuma committed:
MINOR: Introduce `producer.config` property to `ConsoleProducer`

This makes it easier to pass security properties in the same way
to `ConsoleConsumer` and `ConsoleProducer`.

Author: Ismael Juma <ismael@juma.me.uk>

Reviewers: Guozhang Wang <wangguoz@gmail.com>, Jun Rao <junrao@gmail.com>

Closes #544 from ijuma/producer-config-in-console-producer

(cherry picked from commit 06d2c78164aca4b20d1d76f2ddef563b115f2b81)
Signed-off-by: Jun Rao <junrao@gmail.com>

hachikuji committed:
KAFKA-2841; safe group metadata cache loading/unloading

Author: Jason Gustafson <jason@confluent.io>

Reviewers: Guozhang Wang <wangguoz@gmail.com>, Jun Rao <junrao@gmail.com>

Closes #530 from hachikuji/KAFKA-2841

(cherry picked from commit dbdec927b9badcdbe4a4b4b6ccebf5044ca32747)
Signed-off-by: Jun Rao <junrao@gmail.com>

junrao committed:
trivial fix to 0.9.0 docs

ewencp committed:
KAFKA-2859: Fix deadlock in WorkerSourceTask.

Author: Ewen Cheslack-Postava <me@ewencp.org>

Reviewers: Gwen Shapira

Closes #554 from ewencp/kafka-2859-deadlock-worker-source-task

(cherry picked from commit 0f00ec97ae328d04c29cb1cb3eabad3f17e31582)
Signed-off-by: Gwen Shapira <cshapi@gmail.com>

hachikuji committed:
KAFKA-2860: better handling of auto commit errors

Author: Jason Gustafson <jason@confluent.io>

Reviewers: Guozhang Wang

Closes #553 from hachikuji/KAFKA-2860

ijuma committed:
MINOR: Documentation improvements

* Fix typo in api.html
* Mark security features as beta quality (similar to new consumer). Is there better wording?
* Improve wording and clarify things in a number of places
* Improve layout of `pre` blocks (tested locally, which doesn't seem to use the same stylesheets as the deployed version)
* Use producer.config in console-producer.sh command
* Improve SASL documentation structure

Author: Ismael Juma <ismael@juma.me.uk>

Reviewers: Jun Rao, Magnus Edenhill, Gwen Shapira

Closes #550 from ijuma/documentation-improvements

(cherry picked from commit c7c7f4cfa7e1c385d5f5706161572d657b495b7a)
Signed-off-by: Gwen Shapira <cshapi@gmail.com>

ijuma committed:
KAFKA-2869; Host used by Authorizer should be IP address not hostname/IP

Author: Ismael Juma <ismael@juma.me.uk>

Reviewers: Jun Rao <junrao@gmail.com>

Closes #567 from ijuma/kafka-2869-host-used-by-authorizer-should-be-ip

(cherry picked from commit d0614f97bc1c45734631770c46ab6a79fc8c8547)
Signed-off-by: Jun Rao <junrao@gmail.com>

ewencp committed:
KAFKA-2867: Fix missing WorkerSourceTask synchronization and handling of InterruptException.

Author: Ewen Cheslack-Postava <me@ewencp.org>

Reviewers: Jun Rao <junrao@gmail.com>

Closes #566 from ewencp/kafka-2867-fix-source-sync-and-interrupt

(cherry picked from commit 7b3d1bf6a398f5b1f454a7719d04f5ee9e630f96)
Signed-off-by: Jun Rao <junrao@gmail.com>

ijuma committed:
KAFKA-2863; Add a `close()` method to `Authorizer`

Author: Ismael Juma <ismael@juma.me.uk>

Reviewers: Jun Rao <junrao@gmail.com>

Closes #568 from ijuma/kafka-2863-authorizer-close

(cherry picked from commit b1d17e7ef0e901234d95d7825f6862d2aaead76f)
Signed-off-by: Jun Rao <junrao@gmail.com>

junrao committed:
trivial change to 0.9.0 docs to fix outdated ConsumerMetadataRequest

junrao committed:
trivial change to 0.9.0 docs to fix incorrect ssl.key.password

rajinisivaram committed:
KAFKA-2718: Avoid reusing temporary directories in core unit tests

Retry to find new directory and cleanup on exit.

Author: Rajini Sivaram <rajinisivaram@googlemail.com>

Reviewers: Ismael Juma, Guozhang Wang

Closes #399 from rajinisivaram/KAFKA-2718

(cherry picked from commit 0d68eb73f3af8d0e8e32a1e61d8a569c615c3c8c)
Signed-off-by: Guozhang Wang <guozhang@Guozhang-Macbook.local>

hachikuji committed:
KAFKA-2812: improve consumer integration tests

Author: Jason Gustafson <jason@confluent.io>

Reviewers: Geoff Anderson

Closes #500 from hachikuji/KAFKA-2812

(cherry picked from commit b16817a54c592eefc5a462132f45c5b4f786d5f1)
Signed-off-by: Guozhang Wang <guozhang@Guozhang-Macbook.local>

junrao committed:
trivial change: revert incorrect change to ssl.key.password

junrao committed:
trivial doc fix to add link to Quotas

hachikuji committed:
KAFKA-2877: handle request timeout in sync group

Author: Jason Gustafson <jason@confluent.io>

Reviewers: Ben Stopford, Guozhang Wang

Closes #582 from hachikuji/KAFKA-2877

(cherry picked from commit 394eda510dfc1af5c614fa88ee96c7cf2c098fd0)
Signed-off-by: Guozhang Wang <wangguoz@gmail.com>

guozhangwang committed:
KAFKA-2881: Improve Consumer Configs and API Documentation

Author: Guozhang Wang <wangguoz@gmail.com>

Reviewers: Gwen Shapira

Closes #584 from guozhangwang/K2881

(cherry picked from commit 395bf46ddac8c5fe1d6fd0d1636b94dcddadd678)
Signed-off-by: Gwen Shapira <cshapi@gmail.com>

rajinisivaram committed:
KAFKA-2878; Guard against OutOfMemory in Kafka broker

Sanity check array size in requests before allocation

Author: Rajini Sivaram <rajinisivaram@googlemail.com>

Reviewers: Ismael Juma <ismael@juma.me.uk>, Ashish Singh <asingh@cloudera.com>, Jun Rao <junrao@gmail.com>

Closes #577 from rajinisivaram/KAFKA-2878

rajinisivaram committed:
KAFKA-2718: Prevent temp directory being reused in parallel test runs

Use Files.createTempDirectory to avoid reuse, for log directories create a new temp directory as parent

Author: Rajini Sivaram <rajinisivaram@googlemail.com>

Reviewers: Ismael Juma, Guozhang Wang

Closes #583 from rajinisivaram/KAFKA-2718-v2

(cherry picked from commit 8ed271b82abd4641b594aff262512bd9eb588263)
Signed-off-by: Guozhang Wang <wangguoz@gmail.com>

eljefe6a committed:
KAFKA-2892 Consumer Docs Use Wrong Method

The KafkaConsumer docs use a non-existent method for assigning partitions (consumer.assign).

The JavaDocs show as:
```
String topic = "foo";
TopicPartition partition0 = new TopicPartition(topic, 0);
TopicPartition partition1 = new TopicPartition(topic, 1);
consumer.assign(partition0);
consumer.assign(partition1);
```

Should be:
```
String topic = "foo";
TopicPartition partition0 = new TopicPartition(topic, 0);
TopicPartition partition1 = new TopicPartition(topic, 1);
consumer.assign(Arrays.asList(partition0, partition1));
```

Author: Jesse Anderson <eljefe6a@gmail.com>

Reviewers: Ewen Cheslack-Postava, Gwen Shapira

Closes #592 from eljefe6a/trunk

(cherry picked from commit c18a1bd64e7f8416e8256b968fe1b9e2da553b00)
Signed-off-by: Gwen Shapira <cshapi@gmail.com>

logarithm committed:
KAFKA-2882: Add constructor cache for Snappy and LZ4 Output/Input streams in Compressor.java

In `wrapForOutput` and `wrapForInput` methods of `org.apache.kafka.common.record.Compressor`,  `Class.forName("[compression codec]")` and `getConstructor` methods are invoked for each `wrapForOutput` / `wrapForInput` call. Reflection calls are expensive and impact performance at high volumes. This patch adds a cache for `Constructor` to reduce the reflection overhead.

In our production deployments, this has reduced producer CPU usage by about 20%

Author: Maksim Logvinenko <mlogvinenko@gmail.com>

Reviewers: Ismael Juma

Closes #580 from logarithm/compressor-getclass-cache

(cherry picked from commit 4a0e011be3d038763d6326bb0092524f809c3f4d)
Signed-off-by: Guozhang Wang <wangguoz@gmail.com>

benstopford committed:
KAFKA-2899: improve logging when unexpected exceptions thrown in reading local log

Currently we don't log exceptions raised when reading from the local log which makes tracking down the cause of problems a bit tricky.

Author: Ben Stopford <benstopford@gmail.com>

Reviewers: Guozhang Wang

Closes #593 from benstopford/small-patches

(cherry picked from commit 09b43beb1d0aabb2776da2a5790752992ef617f7)
Signed-off-by: Guozhang Wang <wangguoz@gmail.com>

nurkiewicz committed:
MINOR: Avoiding warning about generics in sample code

Author: Tomasz Nurkiewicz <nurkiewicz@gmail.com>

Reviewers: Guozhang Wang

Closes #594 from nurkiewicz/patch-1

(cherry picked from commit 1dd558aeceffdd98da83f8240cdc3bf8bc182c93)
Signed-off-by: Guozhang Wang <wangguoz@gmail.com>

ewencp committed:
KAFKA-2906: Fix Connect javadocs, restrict only to api subproject, and clean up javadoc warnings.

Author: Ewen Cheslack-Postava <me@ewencp.org>

Reviewers: Gwen Shapira

Closes #599 from ewencp/kafka-2906-connect-javadocs

(cherry picked from commit 75c7abd82628b7aae026a662ad0395d1f9c8536b)
Signed-off-by: Gwen Shapira <cshapi@gmail.com>

hachikuji committed:
MINOR: fix verifiable consumer assertion

Author: Jason Gustafson <jason@confluent.io>

Reviewers: Guozhang Wang

Closes #597 from hachikuji/fix-verifiable-consumer-assertion

(cherry picked from commit 78e7f87db8be3b1eae59752e8c9ceb0b8c064a96)
Signed-off-by: Guozhang Wang <wangguoz@gmail.com>

Commit:
MINOR: Fixed new consumer Javadoc typo

Changed "an" to "and"

Author: Alex Sherwin <alex-sherwin@users.noreply.github.com>

Reviewers: Guozhang Wang

Closes #601 from alex-sherwin/patch-1

(cherry picked from commit ea971397c8c1b8deedc48c4165aded42118eee43)
Signed-off-by: Guozhang Wang <wangguoz@gmail.com>

ijuma committed:
MINOR: Update "Java Version" section

Use the latest information from LinkedIn and mention that the latest
released version is recommended from a security perspective.

Author: Ismael Juma <ismael@juma.me.uk>

Reviewers: Grant Henke

Closes #603 from ijuma/java-version-recommendation

hachikuji committed:
KAFKA-2913: missing partition check when removing groups from cache

Author: Jason Gustafson <jason@confluent.io>

Reviewers: Guozhang Wang

Closes #605 from hachikuji/KAFKA-2913

(cherry picked from commit eafc349a1f26a20086b320916c2c819e05bc77ea)
Signed-off-by: Guozhang Wang <wangguoz@gmail.com>

ijuma committed:
MINOR: Update LinkedIn JVM tuning settings

This is a follow-up as the previous update was missing some changes.

Author: Ismael Juma <ismael@juma.me.uk>

Reviewers: Todd Palino, Guozhang Wang

Closes #607 from ijuma/java-tuning-settings

(cherry picked from commit fb6ea6b179b6e4b8a7b23063b17828b42361cd75)
Signed-off-by: Guozhang Wang <wangguoz@gmail.com>

lindong28 committed:
MINOR - fix typo in index corruption warning message

Author: Dong Lin <lindong28@gmail.com>

Reviewers: Ismael Juma, Guozhang Wang

Closes #606 from lindong28/minor-fix-typo

(cherry picked from commit 4649c347b6a59f8fbd4fcaf2363eb172a7da1d4c)
Signed-off-by: Guozhang Wang <wangguoz@gmail.com>

hachikuji committed:
KAFKA-2880: consumer should handle disconnect/timeout for metadata requests

Author: Jason Gustafson <jason@confluent.io>

Reviewers: Ismael Juma, Guozhang Wang

Closes #581 from hachikuji/KAFKA-2880

(cherry picked from commit 9fb1e25738f89b75a36ef69f730b0e138ccd55b1)
Signed-off-by: Guozhang Wang <wangguoz@gmail.com>

Commit:
Manually ported changes in 8c3c9548b636cdf760d2537afe115942d13bc003 to 0.9.0 (no cherry-pick due to conflicts)

luafran committed:
MINOR: ConsoleConsumer - Fix number of processed messages count

kafka-console-consumer.sh is showing an incorrect number of
messages processed, counting one more message than the actual
number of processed messages.

Author: Luciano Afranllie <luafran@gmail.com>

Reviewers: Guozhang Wang

Closes #617 from luafran/console-consumer-number-of-processed-messages

(cherry picked from commit b09663eeec2153e2556983b8ef21551e69ee6afc)
Signed-off-by: Guozhang Wang <wangguoz@gmail.com>

hachikuji committed:
KAFKA-2942: inadvertent auto-commit when pre-fetching can cause message loss

Author: Jason Gustafson <jason@confluent.io>

Reviewers: Guozhang Wang

Closes #623 from hachikuji/KAFKA-2942

(cherry picked from commit 13e483adeee8d968397a21bde3bb159516f26ff0)
Signed-off-by: Guozhang Wang <wangguoz@gmail.com>

granthenke commented:
@chamberlain1990 Did you mean to open this pull request? It looks like its against the wrong branch and has no descriptive message or comment. Could you close this and re-open when fixed?

hachikuji committed:
MINOR: backport fix to partition assignor order (KAFKA-2931)

Author: Jason Gustafson <jason@confluent.io>

Reviewers: Guozhang Wang, Gwen Shapira

Closes #630 from hachikuji/KAFKA-2931-0.9

jkreps committed:
KAFKA-2950: Fix performance regression in the producer

Removes all the System.currentTimeMillis calls to help with performance on small messages.

Author: Jay Kreps <jay.kreps@gmail.com>

Reviewers: Guozhang Wang

Closes #632 from jkreps/producer-perf-regression

(cherry picked from commit 23f36c5903e9b65c2b322a9405149de671de7a23)
Signed-off-by: Guozhang Wang <wangguoz@gmail.com>

granthenke committed:
MINOR: Remove unused DoublyLinkedList

It used to be used by MirrorMaker but its usage was removed in KAFKA-1997.

Author: Grant Henke <granthenke@gmail.com>

Reviewers: Jiangjie Qin

Closes #638 from granthenke/remove-dll

(cherry picked from commit b390b15cdbf89f7ffce16dad2a20ee9330f82731)
Signed-off-by: Guozhang Wang <wangguoz@gmail.com>

vahidhashemian committed:
KAFKA-2957: Fix typos in Kafka documentation

Author: Vahid Hashemian <vahidhashemian@us.ibm.com>

Reviewers: Gwen Shapira

Closes #641 from vahidhashemian/KAFKA-2957

hachikuji committed:
KAFKA-2924: support offsets topic in DumpLogSegments

Author: Jason Gustafson <jason@confluent.io>

Reviewers: Ismael Juma, Guozhang Wang

Closes #622 from hachikuji/KAFKA-2924

(cherry picked from commit 454d7d09014c0e6be3064a120c80261998341fed)
Signed-off-by: Guozhang Wang <wangguoz@gmail.com>

ijuma committed:
KAFKA-2973; Fix leak of child sensors on remove

Author: Ismael Juma <ismael@juma.me.uk>

Reviewers: Gwen Shapira, Guozhang Wang

Closes #650 from ijuma/kafka-2973-fix-leak-child-sensors-on-remove

(cherry picked from commit 977ebbe9bafb6c1a6e1be69620f745712118fe80)
Signed-off-by: Gwen Shapira <cshapi@gmail.com>

ijuma committed:
KAFKA-2972; Add missing `partitionsRemaingList.add` in `ControlledShutdownResponse` constructor

Author: Ismael Juma <ismael@juma.me.uk>

Reviewers: Grant Henke, Guozhang Wang

Closes #649 from ijuma/KAFKA-2972-controlled-shutdown-response-bug

(cherry picked from commit 7bb7e415a371909757a6099a911587047d4c4b69)
Signed-off-by: Guozhang Wang <wangguoz@gmail.com>

eribeiro committed:
KAFKA-2974; `==` is used incorrectly in a few places in Java code

A few issues found via static analysis.

Author: Edward Ribeiro <edward.ribeiro@gmail.com>
Author: Ismael Juma <ismael@juma.me.uk>

Reviewers: Gwen Shapira, Sriharsha Chintalapani, Guozhang Wang

Closes #652 from ijuma/use-equals-instead-of-==

(cherry picked from commit 6e5bd2497af2b6bb67a47e4f3defa50da3cad4c7)
Signed-off-by: Guozhang Wang <wangguoz@gmail.com>

eribeiro committed:
KAFKA-2578; Client Metadata internal state should be synchronized

Author: Edward Ribeiro <edward.ribeiro@gmail.com>

Reviewers: Jason Gustafson <jason@confluent.io>, Ismael Juma <ismael@juma.me.uk>, Ewen Cheslack-Postava <ewen@confluent.io>

Closes #659 from ijuma/KAFKA-2578

(cherry picked from commit ed8748b7d17448a215ebbf68749d311aae30d7f7)
Signed-off-by: Ewen Cheslack-Postava <me@ewencp.org>

Commit:
KAFKA-2893: Add a simple non-negative partition seek check

Author: jinxing <jinxing@fenbi.com>
Author: ZoneMayor <jinxing6042@126.com>

Reviewers: Guozhang Wang

Closes #628 from ZoneMayor/trunk-KAFKA-2893

(cherry picked from commit 9d23b512ccb95940782f51c93bd6cdbed3e938db)
Signed-off-by: Guozhang Wang <wangguoz@gmail.com>

gwenshap committed:
KAFKA-2926; [MirrorMaker] InternalRebalancer calls wrong method of external rebalancer

…ternal rebalancer

Author: Gwen Shapira <cshapi@gmail.com>

Reviewers: Ismael Juma <ismael@juma.me.uk>, Sriharsha Chintalapani <harsha@hortonworks.com>, Ewen Cheslack-Postava <ewen@confluent.io>

Closes #611 from gwenshap/KAFKA-2926

(cherry picked from commit 882fe01284ef7d61e19e4a527fb7c18471a8c074)
Signed-off-by: Ewen Cheslack-Postava <me@ewencp.org>

vesense committed:
KAFKA-2981: Fix javadoc in KafkaConsumer

https://issues.apache.org/jira/browse/KAFKA-2981

Author: Xin Wang <best.wangxin@163.com>

Reviewers: Guozhang Wang

Closes #668 from vesense/patch-2

(cherry picked from commit fd8af25d0ae354e962850c9d6b5991056613aa2d)
Signed-off-by: Guozhang Wang <wangguoz@gmail.com>

hachikuji committed:
KAFKA-2978: consumer stops fetching when consumed and fetch positions get out of sync

Author: Jason Gustafson <jason@confluent.io>

Reviewers: Michal Turek, Ismael Juma, Guozhang Wang

Closes #666 from hachikuji/KAFKA-2978

(cherry picked from commit e08b922aad485a96fceb51d5c27877729156ddab)
Signed-off-by: Guozhang Wang <wangguoz@gmail.com>

hachikuji committed:
KAFKA-2990; Fix NoSuchMethodError in Pool with cast to ConcurrentMap

Author: Jason Gustafson <jason@confluent.io>

Reviewers: Ismael Juma <ismael@juma.me.uk>, Ewen Cheslack-Postava <ewen@confluent.io>

Closes #675 from hachikuji/KAFKA-2990

(cherry picked from commit f3fc9e108eb1e5d47144cc7f2eaa8beb4f88be6b)
Signed-off-by: Ewen Cheslack-Postava <me@ewencp.org>

edenhill committed:
MINOR: Documentation updates for reserved.broker.max.id

People are facing problems upgrading their clusters with configured broker IDs above 1000 due to `reserved.broker.max.id` which wasn't very well announced.

This PR attempts to improve that somewhat by fixing the broker config docs and adding a note to the upgrade documentation.

Author: Magnus Edenhill <magnus@edenhill.se>

Reviewers: Grant Henke <granthenke@gmail.com>, Ewen Cheslack-Postava <ewen@confluent.io>

Closes #670 from edenhill/docs-reserved.broker.max.id

(cherry picked from commit 9a0e16397e81323a3c93053ee473dfcdf3bcceb3)
Signed-off-by: Ewen Cheslack-Postava <me@ewencp.org>

ijuma committed:
MINOR: Add information to upgrade notes

Credit to Gwen for some of the text.

Author: Ismael Juma <ismael@juma.me.uk>

Reviewers: Guozhang Wang <wangguoz@gmail.com>, Ewen Cheslack-Postava <ewen@confluent.io>

Closes #678 from ijuma/mirror-maker-compatibility-note

(cherry picked from commit 4ad165c0783bc49e5750292cd0e8d277e3186846)
Signed-off-by: Ewen Cheslack-Postava <me@ewencp.org>

Commit:
KAFKA-2058: Fix ProducerTest.testSendWithDeadBroker transient failure

It turns that waitUntilMetadataIsPropagated is not enough;
in "onBrokerStartup", methods below will send send both LeaderAndIsrRequest and UpdateMetadataRequest to KafkaApis:
    replicaStateMachine.handleStateChanges(allReplicasOnNewBrokers, OnlineReplica)
    partitionStateMachine.triggerOnlinePartitionStateChange()
The two kinds of request are handled seperately and we are not sure about the order;
If UpdateMetadataRequest is handled first, metadataCache of kafkaApis will be updated, thus TestUtils.waitUntilMetadataIsPropagated will be satisfied, and consumer can(will) start fetching data;
But if the LeaderAndIsrRequest is not handled at this moment, "becomeLeaderOrFollower" cannot be called , thus structures like "leaderReplicaOpt" cannot be updated, which leads to failure of consumer's fetching data;
To fix above, consumer should start fetching data after partition's leaderReplica is refreshed, not just the leader is elected;
So added "TestUtils.waitUntilLeaderIsKnown(servers, topic, 0)"

Author: ZoneMayor <jinxing6042@126.com>
Author: jinxing <jinxing@fenbi.com>

Reviewers: Ismael Juma, Guozhang Wang

Closes #689 from ZoneMayor/trunk-KAFKA-2058

(cherry picked from commit 3615e4773dd20327dead494cd0ba6a26edada92d)
Signed-off-by: Guozhang Wang <wangguoz@gmail.com>

Commit:
KAFKA-2977: Transient Failure in kafka.log.LogCleanerIntegrationTest.cleanerTest

Make MinCleanableDirtyRatioProp configurable(default 0.0F)in makeCleaner, thus log cleaning is always undergoing;
Also removed minDirtyMessages.

Author: jinxing <jinxing@fenbi.com>
Author: ZoneMayor <jinxing6042@126.com>

Reviewers: Ismael Juma, Guozhang Wang

Closes #671 from ZoneMayor/trunk-KAFKA-2977

(cherry picked from commit d2632d011f7bda3cbd0cbcc8f2a3bd1e985d5b1b)
Signed-off-by: Guozhang Wang <wangguoz@gmail.com>

Commit:
KAFKA-2455: Fix failure in kafka.consumer.MetricsTest.testMetricsLeak

DelayedFetchMetrics are loaded dynamically in function "onExpiration";
use assertNull(DelayedFetchMetrics) to initialize DelayedFetchMetrics explicitly;

Author: jinxing <jinxing@fenbi.com>
Author: ZoneMayor <jinxing6042@126.com>

Reviewers: Ismael Juma, Guozhang Wang

Closes #694 from ZoneMayor/trunk-KAFKA-2455

(cherry picked from commit e32131ed6fcfaea7e56295ea0b6ddbf8c804460f)
Signed-off-by: Guozhang Wang <wangguoz@gmail.com>

hachikuji committed:
KAFKA-3014: fix integer overflow problem in leastLoadedNode

Author: Jason Gustafson <jason@confluent.io>

Reviewers: Guozhang Wang

Closes #696 from hachikuji/KAFKA-3014

(cherry picked from commit fa4ffb8767da18b2b84ef3d3cbeb2d4c5b804dbe)
Signed-off-by: Guozhang Wang <wangguoz@gmail.com>

vahidhashemian committed:
MINOR: Fix typo in documentation

Author: Vahid Hashemian <vahidhashemian@us.ibm.com>

Reviewers: Guozhang Wang

Closes #707 from vahidhashemian/typo04/fix_documentation_typos

(cherry picked from commit a6889a2bf94cb326247b5076f474337fbadfadf4)
Signed-off-by: Guozhang Wang <wangguoz@gmail.com>

gwenshap commented:
@chamberlain1990 please close this pull request.

sasakitoa committed:
KAFKA-3002; Allow uppercase letters in hostnames

Make available to specify hostname with Uppercase at broker list

Author: Sasaki Toru <sasakitoa@nttdata.co.jp>

Reviewers: Ismael Juma <ismael@juma.me.uk>, Grant Henke <granthenke@gmail.com>, Ewen Cheslack-Postava <ewen@confluent.io>

Closes #685 from sasakitoa/hostname_uppercase

(cherry picked from commit ea73356905ff4663d0d085c46b348fa4e9b9568d)
Signed-off-by: Ewen Cheslack-Postava <me@ewencp.org>

29x10 committed:
Minor: fix sasl.kerberos.service.name

sasl.kerberos.service.name surround by double quote didn't work, have to remove.

Author: BINLEI XUE <kongpo0412@gmail.com>

Reviewers: Gwen Shapira

Closes #720 from secjex/patch-1

Commit:
KAFKA-3055; Fix JsonConverter mangling the Schema in Connect

Author: ksenji <ksenji@ebay.com>

Reviewers: Dong Lin <lindong28@gmail.com>, Ewen Cheslack-Postava <ewen@confluent.io>

Closes #722 from ksenji/trunk

(cherry picked from commit 2679524604b611046e9826b2a1fba461d42f06f4)
Signed-off-by: Ewen Cheslack-Postava <me@ewencp.org>

ijuma committed:
KAFKA-3052; Broker properties get logged twice if acl enabled

Fix it by making it possible to pass the `doLog` parameter to `AbstractConfig`. As explained in the code comments, this means that we can continue to benefit from ZK default settings as specified in `KafkaConfig` without having to duplicate code.

Also:
* Removed unused private methods from `KafkaConfig`
* Removed `case` modifier from `KafkaConfig` so that `hashCode`, `equals`
and `toString` from `AbstractConfig` are used.
* Made `props` a `val` and added `apply` method to `KafkaConfig` to
remain binary compatible.
* Call authorizer.close even if an exception is thrown during `configure`.

Author: Ismael Juma <ismael@juma.me.uk>

Reviewers: Guozhang Wang

Closes #725 from ijuma/kafka-3052-broker-properties-get-logged-twice-if-acl-enabled

(cherry picked from commit f9642e2a9878faff81366dbc885a206727bd7c7b)
Signed-off-by: Guozhang Wang <wangguoz@gmail.com>

ewencp committed:
MINOR: Update version to 0.9.0.1-SNAPSHOT

Author: Ewen Cheslack-Postava <me@ewencp.org>

Reviewers: Gwen Shapira

Closes #727 from ewencp/minor-0.9.0.1-SNAPSHOT

ijuma committed:
KAFKA-3051 KAFKA-3048; Security config docs improvements

Author: Ismael Juma <ismael@juma.me.uk>

Reviewers: Jun Rao <junrao@gmail.com>

Closes #724 from ijuma/minor-security-fixes

(cherry picked from commit 57df460f8d7b225509dd8c061a5b6efa65c8ac9c)
Signed-off-by: Jun Rao <junrao@gmail.com>

ijuma committed:
MINOR: Add secondary constructor that takes only `props` in `KafkaConfig`

This fixes a compatibility break for users that use `new KafkaConfig(map)`
instead of `KafkaConfig(map)`.

Author: Ismael Juma <ismael@juma.me.uk>

Reviewers: Ewen Cheslack-Postava <ewen@confluent.io>

Closes #728 from ijuma/add-secondary-kafka-config-constructor

ijuma committed:
KAFKA-3010; Include error in log when ack 0

I verified this by trying to produce to __consumer_offsets and the logged message looks like:

[2015-12-22 10:34:40,897] INFO [KafkaApi-0] Closing connection due to error during produce request with correlation id 1 from client id console-producer with ack=0
Topic and partition to exceptions: [__consumer_offsets,43] -> kafka.common.InvalidTopicException (kafka.server.KafkaApis)

Author: Ismael Juma <ismael@juma.me.uk>

Reviewers: Jun Rao <junrao@gmail.com>

Closes #709 from ijuma/kafka-3010-include-error-in-log-when-ack-0

(cherry picked from commit 5db147c1d453c9dabcc277bd95435f17201b9c1c)
Signed-off-by: Jun Rao <junrao@gmail.com>

lindong28 committed:
KAFKA-3045; ZkNodeChangeNotificationListener shouldn't log InterruptedException as error

Author: Dong Lin <lindong28@gmail.com>

Reviewers: Ismael Juma <ismael@juma.me.uk>, Jun Rao <junrao@gmail.com>

Closes #731 from lindong28/KAFKA-3045

(cherry picked from commit c44c898ec835bb16950137f831cbc18c6b8b82f5)
Signed-off-by: Jun Rao <junrao@gmail.com>

Commit:
KAFKA-2937; Disable the leaderIsr check if the topic is to be deleted.

The check was implemented in KAFKA-340 : If we are shutting down a broker when the ISR of a partition includes only that broker, we could lose some messages that have been previously committed. For clean shutdown, we need to guarantee that there is at least 1 other broker in ISR after the broker is shut down.

When we are deleting the topic, this check can be avoided.

Author: Mayuresh Gharat <mgharat@mgharat-ld1.linkedin.biz>

Reviewers: Dong Lin <lindong28@gmail.com>, Rajini Sivaram <rajinisivaram@googlemail.com>, Jun Rao <junrao@gmail.com>

Closes #729 from MayureshGharat/kafka-2937

(cherry picked from commit a788c65f02ae493abc27fe63c6d181e6a9b8edfe)
Signed-off-by: Jun Rao <junrao@gmail.com>

granthenke committed:
KAFKA-2988; Change default configuration of the log cleaner

Author: Grant Henke <granthenke@gmail.com>

Reviewers: Ismael Juma <ismael@juma.me.uk>, Jun Rao <junrao@gmail.com>

Closes #686 from granthenke/compaction

(cherry picked from commit ee1770e00e841ba68dc25f03a8d3f2ac647b0eb3)
Signed-off-by: Jun Rao <junrao@gmail.com>

ijuma committed:
MINOR: Security doc fixes

Simple fixes that have tripped users.

Author: Ismael Juma <ismael@juma.me.uk>

Reviewers: Ewen Cheslack-Postava <ewen@confluent.io>

Closes #745 from ijuma/security-doc-improvements

(cherry picked from commit 36f5c46a5cf510001a4990db6199beb37f215007)
Signed-off-by: Ewen Cheslack-Postava <me@ewencp.org>

taxiao committed:
KAFKA-2993: Calculate compression rate at close() call

Buffer is rewound before the compression rate metric is updated which results in 0 compress rate. The fix is to include a new compressRate field to record the latest compression rate in ```close()``` and return it to sensor

Author: Xiao, Tao <taxiao@ebay.com>

Reviewers: Guozhang Wang

Closes #784 from xiaotao183/kafka-2993

(cherry picked from commit a071e3554bf626182557dfaca491afebe9c17020)
Signed-off-by: Guozhang Wang <wangguoz@gmail.com>

hachikuji committed:
KAFKA-2695: limited support for nullable byte arrays

Author: Jason Gustafson <jason@confluent.io>

Reviewers: Guozhang Wang

Closes #780 from hachikuji/KAFKA-2695

(cherry picked from commit 747dc930ff710ec890b5f054b4ec4bf7bbb7c960)
Signed-off-by: Guozhang Wang <wangguoz@gmail.com>

dajac committed:
KAFKA-3098: "partition.assignment.strategy" appears twice in documentation

Author: David Jacot <david.jacot@gmail.com>

Reviewers: Gwen Shapira

Closes #774 from dajac/KAFKA-3098

(cherry picked from commit 5979480c3af6be00cf7b135aeabf11bf8141f936)
Signed-off-by: Gwen Shapira <cshapi@gmail.com>

Commit:
KAFKA-3095: Add documentation on format of sasl.kerberos.principal.to.local.rules

Add some basic documentation about the format, a link to get more detailed information and an example usage.  I didn't want to make a huge section on the format since it documented elsewhere but I can expand is folks want.

https://issues.apache.org/jira/browse/KAFKA-3095

Author: Tom Graves <tgraves@yahoo-inc.com>

Reviewers: Gwen Shapira

Closes #776 from tgravescs/KAFKA-3095

(cherry picked from commit 401ae7707cf3990240544eb4575acc46a9c85017)
Signed-off-by: Gwen Shapira <cshapi@gmail.com>

granthenke committed:
KAFKA-3012: Avoid reserved.broker.max.id collisions on upgrade

Provides a configuration to opt out of broker id generation.

Author: Grant Henke <granthenke@gmail.com>

Reviewers: Gwen Shapira

Closes #762 from granthenke/id-generation

(cherry picked from commit 5c337d759892ae6d46d6901a24e9b97cebd2a4da)
Signed-off-by: Gwen Shapira <cshapi@gmail.com>

dajac committed:
MINOR: MemoryRecords.sizeInBytes throws NPE when non-writable.

I just noticed that `MemoryRecords.sizeInBytes` throws NPE when MemoryRecords is non-writable. `compressor` is explicitly set to null when `writable` is false (L56) at the construction time, for instance when `MemoryRecords.readableRecords` is used.

guozhangwang Could you take a look when you have time?

Author: David Jacot <david.jacot@gmail.com>

Reviewers: Guozhang Wang

Closes #786 from dajac/kafka-npe

(cherry picked from commit 0a7f2bf335e426b12f717ec7f6d51779d56fe59b)
Signed-off-by: Guozhang Wang <wangguoz@gmail.com>
